### PR TITLE
Add a tag to sample_correlation and MultiQC processes

### PR DESCRIPTION
**Pull request by Hammarn**
This borrows the expression from the STAR process and uses the basename of the first file in the list as a tag for these processes. 

This is based on the NGI naming convention, but I think that should be fine. Either the user will end up with nothing as the tag (currently the case) or with the basename, which should be fine. 
I haven't tested it though since our cluster is down a the moment.

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -711,6 +712,8 @@ process sample_correlation {
 
     script: // This script is bundled with the pipeline, in NGI-RNAseq/bin/
     def rlocation = params.rlocation ?: ''
+    prefix = reads[0].toString() - ~/(_R1)?(_trimmed)?(_val_1)?(\.fq)?(\.fastq)?(\.gz)?P\d+(_\d+)?$/
```
Comment (ewels) [inline main.nf]: Isn't this _removing_ the `P1234` number? The `reads[0].toString() - 'xxx'` removes `'xxx'` from the string. So by adding `P\d+(_\d+)?` to the end of the regex, I think you're removing that from the basename?

**Comment by Hammarn** [comment]
I was expecting it to remove the capture group. I .e the sample number `(_\
d+)`

**Comment by ewels** [comment]
The brackets are there to act as groupings - the `?` makes the match of anything inside the brackets optional. Anything that matches inside the regex will then be deleted. They're not being used as capture groups if that's what you mean.. You'd need to use a different groovy function for that. Plus I think it's better to delete away to the basename rather than try to capture the NGI id, as then it will work for other people.

**Comment by ewels** [comment]
Nice! Can you add changelog comments about these additions in the future too please.. :)

**Pull request merged**